### PR TITLE
MISUV-6507 Remove duplicate Unit tests for Income Source Journeys 1

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 1]
```diff
@@ -81,536 +82,367 @@ class AddBusinessNameControllerSpec extends TestSupport
...
-      setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-
-      val result: Future[Result] = TestAddBusinessNameController.submitChangeAgent()(fakePostRequestConfirmedClient().withFormUrlEncodedBody(
-        BusinessNameForm.businessName -> invalidBusinessNameEmpty
-      ))
-
-      status(result) mustBe BAD_REQUEST
-      contentAsString(result) must include("Enter your name or the name of your business")
+      //
+      //      s"Change - AddBusinessNameController.submitChange ${if (isAgent) "Agent" else "Individual"}" should {
```
Comment (simondrugan16) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:344]: ❓ Are these commented lines supposed to be here?

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 2]
```diff
@@ -81,536 +82,367 @@ class AddBusinessNameControllerSpec extends TestSupport
...
-          val redirectUrl = controllers.incomeSources.add.routes.IncomeSourceCheckDetailsController.show(SelfEmployment).url
-
-          val result: Future[Result] = TestAddBusinessNameController.submitChange()(fakePostRequestConfirmedClient().withFormUrlEncodedBody(
-            BusinessNameForm.businessName -> validBusinessName))
-          status(result) shouldBe SEE_OTHER
-          redirectLocation(result) mustBe Some(redirectUrl)
-          verify(mockSessionService)
-            .setMongoKey(ArgumentMatchers.eq(businessNameField), ArgumentMatchers.eq(validBusinessName), ArgumentMatchers.eq(journeyType))(any(), any())
+      s"ADD - AddBusinessNameController.submit (${if (isAgent) "Agent" else "Individual"}, isChange = $isChange)" should {
+        "return 303 and redirect to add business start date" when {
```
Comment (simondrugan16) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:211]: ❓ Are you testing all scenarios for isAgent and isChange here? I can't spot where you're running through the inputs

File: test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala [hunk 3]
```diff
@@ -83,340 +84,231 @@ class AddBusinessTradeControllerSpec extends TestSupport
...
+  }
+  def postRequest(isAgent: Boolean) = {
+    if (isAgent) fakePostRequestConfirmedClient()
+    else fakePostRequestWithActiveSession
+  }
 
-        val result: Future[Result] = TestAddBusinessTradeController.show(isAgent = false, isChange = false)(fakeRequestWithActiveSession)
-        status(result) shouldBe OK
+  for (isAgent <- Seq(true, false)) yield {
+    s"AddBusinessTradeController ${if (isAgent) "Agent" else "Individual"}" should {
```
Comment (simondrugan16) [inline test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala:102]: 💭 Change to AddBusinessTradeController.show in matcher to make more clear perhaps

File: test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala [hunk 4]
```diff
@@ -83,340 +84,231 @@ class AddBusinessTradeControllerSpec extends TestSupport
...
+      "redirect a user back to the custom error page" when {
+        "the individual is not authenticated" should {
+          "redirect them to sign in" in {
+            if (isAgent) setupMockAgentAuthorisationException() else setupMockAuthorisationException()
+            val result = TestAddBusinessTradeController.show(isAgent, isChange = false)(getRequest(isAgent))
```
Comment (simondrugan16) [inline test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala:107]: ⛏️ Why no isChange = false test?

File: test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala [hunk 5]
```diff
@@ -38,9 +38,10 @@ import play.api.mvc.{MessagesControllerComponents, Result}
 import play.api.test.Helpers._
 import services.IncomeSourceDetailsService
 import testConstants.BaseTestConstants
-import testConstants.BaseTestConstants.{testAgentAuthRetrievalSuccess, testSessionId}
+import testConstants.BaseTestConstants.{testAgentAuthRetrievalSuccess, testIndividualAuthSuccessWithSaUtrResponse, testSessionId}
 import testConstants.incomeSources.IncomeSourceDetailsTestConstants.businessesAndPropertyIncome
```
Comment (simondrugan16) [inline test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala:42]: ❓ does this spec have tests for the trade and name being the specifically for isChange = true pages?

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala (hunk 2 above)
Comment (danielmaltz1) [reply test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:211]: There's a 
for (isAgent <- Seq(true, false)) yield {
    for (isChange <- Seq(true, false)) yield {
around the whole thing, so it essentially runs the whole thing 4 times with the different scenarios

File: test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala (hunk 4 above)
Comment (danielmaltz1) [reply test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala:107]: There were no tests for the change case there to begin with. I was going to create a tech debt to add them once this ticket was done.

**Comment by avelx** [comment]
@simondrugan16 / @danielmaltz1  - relevant story can be moved to done as there are no a changes in the main logic;

**Pull request merged**